x/bank/client/cli/query.go

```diff
@@ -43,35 +45,47 @@ func GetBalancesCmd(clientCtx client.Context) *cobra.Command {
 		Short: "Query for account balances by address",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			queryClient := types.NewQueryClient(clientCtx.Init())
+			height, err := cmd.Flags().GetInt64(flags.FlagHeight)
+			if err != nil {
+				return err
+			}
```

Comment:
So `FlagHeight` is applied by the `flags.GetCommands` middleware (which I think is poorly named - maybe `AddGetCommandFlags`). What would make sense here is a function that works in correspondence with that middleware like: `func ReadGetCommandFlags(client.Context, flag.FlagSet) client.Context`

Reply:
Yeah middleware would be great. I'll think more on this.

Reply:
A couple thoughts.

One, I think somewhere we want to pass a viper instance into the command and mindless handler. Maybe on `client.Context`??

And, what about a more radical approach where `client.Context` gets attached to `cobra.Command.Context()`? Rather than be passed in the command constructor

Reply:
Mhhh I don't how or why we'd want to pass a Viper literal around. I think the current suggestion makes sense.

I also don't see how the more radical approach improves anything either. I don't think having each command handler call `ReadGetCommandFlags()` is that big of a deal IMHO.

Reply:
So `ReadGetCommandFlags` is pretty verbose and `ReadPostCommandFlags` will be even more verbose. Was your followup suggestion along the lines of avoiding this somehow? I guess I understand that's where passing a Viper literal around makes sense. I'll poke around and see how much work that requires.

Reply:
Do you mean that the implementation of `ReadGetCommandFlags` is pretty verbose or that calling it is verbose @alexanderbez ?

Reply:
The implementation. Calling it is fine IMHO.

Reply:
I wonder if there’s a more concise way than calling `Changed()` everywhere but it’s not that bad either. I mostly think this approach makes sense. Maybe there are cleaner ways to do things... not sure really but this approach makes sense.

Reply:
Agree. Ok I'll continue with this for now.